src/DotLiquid/Context.cs

```diff
@@ -42,6 +42,11 @@ public int MaxIterations
 
         private Strainer _strainer;
 
+        /// <summary>
+        /// A cancellation token which indicates if the template render has been cancelled
+        /// </summary>
+        public CancellationToken CancellationToken => _cancellationToken;

```

Comment:
Without implementing a CancellationTokenSource for timeout based cancellations, this is very limited in real use. Adding either or to the RenderParameters doesn't negate that. @daviburg  the suggestion was to change Context to inherit IDisposable so he can use my code snippet safely, but author said it would be a breaking change. What do you suggest?

Another alternative is dropping timeout from Context entirely (it is marked obsolete) and converting the timeout to a CancellationToken in RenderParameters.